it/src/test/java/com/google/cloud/teleport/it/spanner/SpannerResourceManagerUtilsTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.spanner;
+
+import static com.google.cloud.teleport.it.spanner.SpannerResourceManagerUtils.generateDatabaseId;
+import static com.google.cloud.teleport.it.spanner.SpannerResourceManagerUtils.generateInstanceId;
+import static com.google.cloud.teleport.it.spanner.SpannerResourceManagerUtils.generateNewId;
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link com.google.cloud.teleport.it.spanner.SpannerResourceManagerUtils}. */
+@RunWith(JUnit4.class)
+public class SpannerResourceManagerUtilsTest {
```

Comment:
What about test cases for other characters present in the input id?
For example, I imagine resource manager users may be passing the Java class name of the test as the testId, which may contain `.` or `$` chars, which are not allowed in Spanner IDs.

Reply:
I have updated the logic for both generateInstanceId and generateDatabaseId to account for those. For other characters not allowed in Spanner IDs, I'll leave it to Spanner to throw an error